.github/workflows/python.yml

```diff
@@ -39,45 +56,95 @@ permissions:
+          - name: Load the engine matrix
+            id: load-engine-matrix
+            shell: bash
+            # We want to run the full matrix only for scheduled runs and Core changes
+            # For other events we want to run only the always matrix which is first and last version supported
+            run: |
+              if [[ "${{ github.event_name }}" == "pull_request" || "${{github.event_name }}" == "push" ]]; then
+                echo "matrix=$(jq -c '[.[] | select(.run == "always")]' < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
+              else
+                echo "matrix=$(jq -c . < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
+              fi
+
+    load-host-matrix:
+      runs-on: ubuntu-latest
+      needs: check_should_run
+      if: ${{ needs.check_should_run.outputs.should_run == 'true' }}
+      outputs:
+          matrix: ${{ steps.load-host-matrix.outputs.matrix }}
+      steps:
+          - name: Checkout
+            uses: actions/checkout@v4
+          - name: load host matrix
+            id: load-host-matrix
+            shell: bash
+            run: |
+              if [[ "${{ github.event_name }}" == "pull_request" || "${{ github.event_name }}" == "push" ]]; then
+                echo 'matrix={"include":' $(jq '[.[] | select(.run | type == "array" and contains(["always"]))]' .github/json_matrices/build-matrix.json) '}' >> $GITHUB_OUTPUT
+              else
+                echo 'matrix={"include":' $(jq '[.[] | select(.run | type == "array" and contains(["pyton"]))]' .github/json_matrices/build-matrix.json) '}' >> $GITHUB_OUTPUT
+              fi
+
+    create-version-matrix:
         runs-on: ubuntu-latest
+        needs: check_should_run
+        if: ${{ needs.check_should_run.outputs.should_run == 'true' }}
         outputs:
-          matrix: ${{ steps.load-engine-matrix.outputs.matrix }}
+            matrix: ${{ steps.create-version-matrix.outputs.matrix }}
         steps:
             - name: Checkout
               uses: actions/checkout@v4
-        
-            - name: Load the engine matrix
-              id: load-engine-matrix
+
+            - name: Create version matrix
+              id: create-version-matrix
+              
               shell: bash
-              run: echo "matrix=$(jq -c . < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
-        
-    test:
+              # We want to run the full matrix only for scheduled runs and Core changes
+              # For other events we want to run only the always matrix which is first and last version supported
+              run: |
+                if [[ "${{ github.event_name }}" == "pull_request" || "${{github.event_name }}" == "push" ]]; then
+                  echo 'matrix={"python":["3.8","3.12"]}' >> $GITHUB_OUTPUT
+                else
+                  echo 'matrix={"python":["3.8","3.9","3.10","3.11","3.12"]}' >> $GITHUB_OUTPUT
+                fi
+
+    test-python-client:
```

Comment:
```suggestion
    test:
```

Reply:
I don't agree here, I think it's easier when there's a name